perf(transformer): use `FxDashMap` for browser query cache

overlookmotel opened the pull request:
Use `FxDashMap` instead of plain `DashMap` for browser query cache.

Inspired by https://github.com/rolldown/rolldown/pull/2971.

overlookmotel commented:
* **#7521** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7521?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7521?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Nov 28, 6:38 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 28, 6:38 AM EST**: A user added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
perf(transformer): use `FxDashMap` for browser query cache (#7521)

Use `FxDashMap` instead of plain `DashMap` for browser query cache.

Inspired by https://github.com/rolldown/rolldown/pull/2971.
